--- a/src/applet-device-ethernet.c
+++ b/src/applet-device-ethernet.c
@@ -67,6 +67,7 @@ ethernet_new_auto_connection (NMDevice *
                               AppletNewAutoConnectionCallback callback,
                               gpointer callback_data)
 {
+	/**
 	NMConnection *connection;
 	NMSettingWired *s_wired = NULL;
 	NMSettingConnection *s_con;
@@ -90,6 +91,7 @@ ethernet_new_auto_connection (NMDevice *
 	nm_connection_add_setting (connection, NM_SETTING (s_con));
 
 	(*callback) (connection, TRUE, FALSE, callback_data);
+	*/
 	return TRUE;
 }
 
@@ -238,8 +240,6 @@ ethernet_add_menu_item (NMDevice *device
 
 		if (g_slist_length (connections))
 			add_connection_items (device, connections, carrier, active, ADD_INACTIVE, menu, applet);
-		else
-			add_default_connection_item (device, carrier, menu, applet);
 	}
 
 	g_slist_free (connections);
